nengo/processes.py

```diff
@@ -260,3 +260,129 @@ def step_presentinput(t):
             return inputs[i % n]
 
         return step_presentinput
+
+
+class Piecewise(Process):
+    """Present a piecewise input with different options for interpolation.
+
+    Given an input of tp = [0, 0.5, 0.75, 1], yp = [[0], [1], [-1], [0]] this will generate a
+    function that returns the values in yp at corresponding time points.
+
+    Elements in tp must be times (floats or ints). The values in yp can be floats for
+    1d or lists for multi-dimensional function. All lists must be of the same length.
+
+    Interpolations on the data points using scipy.interpolation are also supported. The default
+    interpolation is 'zero', which simply creates a piecewise function whose values begin
+    at the specified time points. So the above example would be shortcut for:
+        def function(t):
+            if t < 0.5:
+                return 0
+            elif t < 0.75
+                return 1
+            elif t < 1:
+                return -1
+            else:
+                return 0
+
+    For times before the first specified time, it will default to zero (of
+    the correct length). This means that the above can be simplified to:
+
+        Piecewise([0.5, 0.75, 1], yp = [[1], [-1], [0]])
+
+    Parameters
+    ----------
+    tp : time points of the function where values are specified
```

Comment:
This will have to be updated, as `tp` and `yp` are no longer attributes of the param.